### PR TITLE
add dummy address to extended p2p

### PR DESCRIPTION
**Pull request by paweljakubas**
# Issue Number

# Overview

- [x] I have added public_address dummy value to p2p Jormungandr config
- [x] I have added `getRandomPort` function and extended `genConfigFile` by port arg

# Comments

### CODE REVIEW

File: lib/jormungandr/src/Cardano/Wallet/Jormungandr/Compatibility.hs [hunk 1]
```diff
@@ -228,6 +228,7 @@ genConfigFile stateDir (BaseUrl _ host port path) = object
             [ "messages" .= String "low"
             , "blocks" .= String "normal"
             ]
+        , "public_address" .= String "ta1svy0mwwm7mdwcuj308aapjw6ra4c3e6cygd0f333nvtjzxg8ahdvxlswdf0"
```
Comment (piotr-iohk) [inline lib/jormungandr/src/Cardano/Wallet/Jormungandr/Compatibility.hs]: With that I get:

```
Error in the overall configuration of the node
Error while parsing the node configuration file: p2p.public_address: invalid value: string "ta1svy0mwwm7mdwcuj308aapjw6ra4c3e6cygd0f333nvtjzxg8ahdvxlswdf0", expected Multiaddr (example: /ip4/192.168.0.1/tcp/443) at line 1 column 162
```
I think the value needs to be something like `/ip4/192.168.0.1/tcp/443` after all...

**Review by piotr-iohk — CHANGES_REQUESTED** [review]

File: lib/jormungandr/src/Cardano/Wallet/Jormungandr/Compatibility.hs (hunk 1 above)
Comment (KtorZ) [reply lib/jormungandr/src/Cardano/Wallet/Jormungandr/Compatibility.hs]: Hmm.. Yeah, we are still looking for an address in the sense of an IP address :sweat_smile:

Comment (paweljakubas) [reply lib/jormungandr/src/Cardano/Wallet/Jormungandr/Compatibility.hs]: I corrected this

Comment (paweljakubas) [reply lib/jormungandr/src/Cardano/Wallet/Jormungandr/Compatibility.hs]: but locally see it freezes at 
```
  PORT_02 - Can omit --port when server uses default port (address list)
Ports CLI (explicit) tests
[iohk.cardano-wallet.serve:Notice:ThreadId 9801] [2019-08-05 15:25:19.15 UTC] 10 migrations were applied to the database.
```

Comment (paweljakubas) [reply lib/jormungandr/src/Cardano/Wallet/Jormungandr/Compatibility.hs]: we have already set 8090 port (defaultPort)

Comment (KtorZ) [reply lib/jormungandr/src/Cardano/Wallet/Jormungandr/Compatibility.hs]: Hmmm. You provided the same value as the default one used by the wallet server 😶

What about making the config generator take an extra port param, and find an adequate port in the launch command handler (by querying the system for a random available one)?

File: lib/jormungandr/src/Cardano/Wallet/Jormungandr/Compatibility.hs [hunk 2]
```diff
@@ -228,6 +228,7 @@ genConfigFile stateDir (BaseUrl _ host port path) = object
             [ "messages" .= String "low"
             , "blocks" .= String "normal"
             ]
+        , "public_address" .= String "/ip4/127.0.0.1/tcp/8081"
```
Comment (piotr-iohk) [inline lib/jormungandr/src/Cardano/Wallet/Jormungandr/Compatibility.hs]: What about this?

>What about making the config generator take an extra port param, and find an adequate port in the launch command handler (by querying the system for a random available one)?

I think that putting explicit port may cause some unnecessary troubleshooting on the user side if it interferes with some running service. Especially that the config is automatically generated, and we put this port somewhat silently.

Comment (paweljakubas) [reply lib/jormungandr/src/Cardano/Wallet/Jormungandr/Compatibility.hs]: @piotr-iohk  @KtorZ  yes, I can add parameter port to `genConfigFile` and add function that uses `bindRandomPortTcp` which get available port by binding to socket, then closes the socket and then deliver the port that was engaged in it. But how to add this in   `config.yaml` ???

Comment (piotr-iohk) [reply lib/jormungandr/src/Cardano/Wallet/Jormungandr/Compatibility.hs]: `lib/jormungandr/test/data/jormungandr/config.yaml` -> here we can have `8081` I think, it's used for starting `jormungandr` for some portion of integration tests.

Comment (paweljakubas) [reply lib/jormungandr/src/Cardano/Wallet/Jormungandr/Compatibility.hs]: ok, give me a sec ;-)

Comment (paweljakubas) [reply lib/jormungandr/src/Cardano/Wallet/Jormungandr/Compatibility.hs]: done

**Review by KtorZ — APPROVED** [review]

**Review by piotr-iohk — APPROVED** [review]
:ok_hand:

**Pull request merged**